File: src/main/scala/com/dyrkin/monkeyc/idea/plugin/psi/MonkeyElementFactory.java
package com.dyrkin.monkeyc.idea.plugin.psi;

import com.dyrkin.monkeyc.idea.plugin.MonkeyFile;
import com.dyrkin.monkeyc.idea.plugin.MonkeyFileType;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiFileFactory;

public class MonkeyElementFactory {

  public static MonkeyNamedElement createProperty(Project project, String name) {

    final MonkeyFile file = createFile(project, name);
    return (MonkeyNamedElement) file.getFirstChild();
  }

  public static MonkeyFile createFile(Project project, String text) {
    String name = "dummy.mc";
    return (MonkeyFile) PsiFileFactory.getInstance(project).
        createFileFromText(name, MonkeyFileType.Instance(), text);
  }
}

File: src/com/dyrkin/monkeyc/idea/plugin/MonkeyElementFactory.scala
package com.dyrkin.monkeyc.idea.plugin

import com.dyrkin.monkeyc.idea.plugin.psi.MonkeyId
import com.intellij.openapi.project.Project
import com.intellij.psi.util.PsiTreeUtil
import com.intellij.psi.{PsiFile, PsiFileFactory}


object MonkeyElementFactory {

  def createIdentifierFromText(project: Project, text: String) = {
    val dummyFile = createDummyFile(project, s"var $text;")
    PsiTreeUtil.findChildOfType(dummyFile, classOf[MonkeyId])
  }

  private def createDummyFile(myProject: Project, content: String): PsiFile = {
    val factory = PsiFileFactory.getInstance(myProject)
    val name = "dummy." + MonkeyFileType().getDefaultExtension()
    factory.createFileFromText(name, MonkeyLanguage(), content)
  }
}
